File: LLM-Translations/GPT-4/codenet/C++/Go/s085712108.go
package main

import (
	"fmt"
	"strconv"
)

const MOD = 7007

type QUEUE struct {
	a, b, c, tot int
}

var F = [6][3]int{{1, 1, 0}, {1, 0, 1}, {0, 1, 1}, {2, 0, 0}, {0, 2, 0}, {0, 0, 2}}
var Hash [MOD][]int

func HASH(n int) bool {
	for _, v := range Hash[n%MOD] {
		if n == v {
			return false
		}
	}
	Hash[n%MOD] = append(Hash[n%MOD], n)
	return true
}

func main() {
	var Push QUEUE
	Push.tot = 0
	var que []QUEUE
	fmt.Scanf("%d%d%d", &Push.a, &Push.b, &Push.c)
	que = append(que, Push)

	for len(que) > 0 {
		Fro := que[0]
		if Fro.a == Fro.b && Fro.b == Fro.c {
			fmt.Printf("%d\n", Fro.tot)
			return
		}
		for i := 0; i < 6; i++ {
			Push = Fro
			Push.tot++
			Push.a += F[i][0]
			Push.b += F[i][1]
			Push.c += F[i][2]
			n := strconv.Itoa(Push.a*10000 + Push.b*100 + Push.c)
			if !HASH(int(n)) {
				continue
			}
			que = append(que, Push)
		}
		que = que[1:]
	}
}

// 

File: LLM-Translations/GPT-4/codenet/C++/Java/s085712108.java
import java.util.*;

public class s085712108 {
    static int MOD = 7007;

    static class QUEUE {
        int a, b, c, tot;
    }

    static ArrayList<ArrayList<Integer>> Hash = new ArrayList<>();

    static boolean HASH(int n) {
        for (int i = 0; i < Hash.get(n % MOD).size(); i++)
            if (n == Hash.get(n % MOD).get(i))
                return false;
        Hash.get(n % MOD).add(n);
        return true;
    }

    public static void main(String[] args) {
        for (int i = 0; i < MOD; i++) {
            Hash.add(new ArrayList<>());
        }

        QUEUE Push = new QUEUE();
        Push.tot = 0;
        Queue<QUEUE> que = new LinkedList<>();

        Scanner sc = new Scanner(System.in);
        Push.a = sc.nextInt();
        Push.b = sc.nextInt();
        Push.c = sc.nextInt();
        sc.close();

        que.add(Push);

        int[][] F = {{1, 1, 0}, {1, 0, 1}, {0, 1, 1}, {2, 0, 0}, {0, 2, 0}, {0, 0, 2}};

        while (!que.isEmpty()) {
            QUEUE Fro = que.poll();

            if (Fro.a == Fro.b && Fro.b == Fro.c) {
                System.out.println(Fro.tot);
                return;
            }

            for (int i = 0; i < 6; i++) {
                Push = new QUEUE();
                Push.tot = Fro.tot + 1;
                Push.a = Fro.a + F[i][0];
                Push.b = Fro.b + F[i][1];
                Push.c = Fro.c + F[i][2];

                int n = Push.a * 10000 + Push.b * 100 + Push.c;
                if (!HASH(n)) continue;
                que.add(Push);
            }
        }
    }
}
// 
